lib/src/main/scala/spinal/lib/bus/amba4/axi/sim/Axi4Master.scala

```diff
@@ -293,16 +298,15 @@ case class Axi4Master(axi: Axi4, clockDomain: ClockDomain, name: String = "unnam
         wQueue += { w =>
           val data = paddedData.slice(beat * bytePerBeat, (beat + 1) * bytePerBeat)
           w.data #= data.toArray
-          val strb = if (len == 0) {
-            ((BigInt(1) << realLen) - 1) << padFront
-          } else beat match {
-            case 0 => (BigInt(1) << (bytePerBeat - padFront)) - 1
-            case `len` => ~((BigInt(1) << padBack) - 1)
-            case _ => BigInt(1) << busConfig.bytePerWord
-          }
+          val fullStrb = (BigInt(1) << bytePerBeat) - 1
+          val strb = (beat match {
+            case 0     => fullStrb << padFront
+            case `len` => fullStrb >> padBack
+            case _     => fullStrb
+          }) & fullStrb
```

Comment:
Overall i think there is still two cases which aren't handled (and were not before the pr either ?)

- Single beat burst with address not being word aligned.
- burst with size smaller than word size

@KireinaHoro
Can you double check ?

Reply:
I added a test for that and it was already working in my branch. This was handled by `padData` in line 284.
Did I understand your points correctly?

Reply:
Looks like narrow bursts will be handled in the proposed version (since you use `fullStrb`).

Not so sure about the unaligned address case -- address is rounded at the beginning through `padData`... @Dolu1990 can you elaborate?